docs/source/api_references.rst

```diff
@@ -27,11 +27,17 @@ Core API
 .. autosummary::
     :toctree: api
     :nosignatures:
+    :template: classtemplate.rst
 
-    datamodule
-    decorators
-    hooks
-    lightning
+    ~hooks.CheckpointHooks
+    ~hooks.DataHooks
+    ~hooks.ModelHooks
+    LightningDataModule
+    LightningModule
+    ~mixins.DeviceDtypeModuleMixin
```

Comment:
More sure we should display the mixin under Core API. This would be quite confusing for the readers.
@carmocca